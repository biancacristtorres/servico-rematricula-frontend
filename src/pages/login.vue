<template>
  <v-container fluid fill-height grid-list-xl>
    <v-layout justify-center align-center>
      <v-flex lg4 md6 sm8 xs12>
        <v-card class="pa-5" raised>
          <v-form>
            <v-alert
              :value="possuiErroCredenciais"
              transition="scale-transition" 
              color="error"
              class="my-4"
              icon="warning"
              outline>
              O R.A. ou senha preenchidos n√£o foram encontrados no sistema. Favor corrigir e tentar novamente.
            </v-alert>

            <v-text-field 
              @keyup.enter="entrar" 
              :error="erros.usuario" 
              data-vv-name="usuario" 
              v-validate="'required'"
              v-model="login"
              name="usuario" 
              label="R.A."
              autocomplete="username" 
              id="login-usuario">
            </v-text-field>

            <v-text-field 
              @keyup.enter="entrar" 
              :error="erros.senha" 
              data-vv-name="senha"
              v-validate="'required'"
              v-model="senha"
              name="senha"
              label="Senha"
              type="password"
              autocomplete="current-password"
              id="login-senha">
            </v-text-field>

            <v-card-actions class="mt-4 justify-center pa-0">
              <v-btn :loading="loading" class="primary" block large @click="entrar" id="login-botao">
                Entrar
              </v-btn>
            </v-card-actions>
          </v-form>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script lang="ts">
import {
  Component,
  Prop,
  Vue,
} from 'vue-property-decorator';

import { namespace } from 'vuex-class';
import { StoreNamespaces } from '@/store';
import { AutorizacaoActionTypes } from 'componente-frontend-core/store/autorizacao/actions';

const autorizacaoNamespace = namespace(StoreNamespaces.AUTORIZACAO);

@Component({})
  export default class Login extends Vue {
    public login: string = '';
    public senha: string = '';
    public possuiErroCredenciais = false;
    public loading = false;
    public erros: any = { usuario: false, senha: false };

    @autorizacaoNamespace.Action(AutorizacaoActionTypes.FAZER_LOGIN)
    public fazerLogin!: (dados: any) => Promise<string>;

    private entrar(): void {
      this.resetarEstadosDeErro();

      this.$validator
        .validateAll()
        .then(this.aoValidarFormulario);
    }

    private resetarEstadosDeErro(): void {
      this.possuiErroCredenciais = false;
      this.erros = { usuario: false, senha: false };
    }

    private aoValidarFormulario(estaValido: any): void {
      if (estaValido) {
        this.quandoOFormularioEstaValido();
      } else {
        this.quandoOFormularioEstaInvalido();
      }
    }

    private quandoOFormularioEstaValido(): void {
      this.ligarLoading();
      this.fazerLogin({ usuario: this.login, senha: this.senha })
        .then(this.loginComSucesso)
        .catch(this.falhaNoLogin)
        .finally(this.desligarLoading);
    }

    private quandoOFormularioEstaInvalido(): void {
        this.erros = { usuario: this.errors.has('usuario'), senha: this.errors.has('senha') };
    }

    private loginComSucesso(): void {
      this.$router.push('/');
    }

    private falhaNoLogin(): void {
      this.possuiErroCredenciais = true;
    }

    private ligarLoading() {
      this.loading = true;
    }

    private desligarLoading() {
      this.loading = false;
    }
}
</script>

