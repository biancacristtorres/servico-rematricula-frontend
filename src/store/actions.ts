import { ActionTree } from 'vuex';
import { RootState } from '.';
import { AlunoMutationTypes } from './aluno/mutations';
import { StoreNamespaces } from './namespaces';
import { ResumoMutationTypes } from './resumo/mutations';
import { GlobalMutationTypes } from './mutations';
import { GlobalState } from './state';

export enum GlobalActionTypes {
  LIMPAR_STORE = 'LIMPAR_STORE',
  DEFINIR_LINK_PARA_VOLTAR = 'DEFINIR_LINK_PARA_VOLTAR',
  LIMPAR_LINK_PARA_VOLTAR = 'LIMPAR_LINK_PARA_VOLTAR',
}

const actions: ActionTree<GlobalState, RootState> = {

  async [GlobalActionTypes.LIMPAR_STORE]({ commit }) {
    commit(`${ StoreNamespaces.ALUNO }/${ AlunoMutationTypes.LIMPAR_STORE }`);
    commit(`${ StoreNamespaces.RESUMO }/${ ResumoMutationTypes.LIMPAR_STORE }`);
  },

  async [GlobalActionTypes.DEFINIR_LINK_PARA_VOLTAR]({ commit }, link: string) {
    commit(GlobalMutationTypes.SET_LINK_PARA_VOLTAR, link);
  },

  async [GlobalActionTypes.LIMPAR_LINK_PARA_VOLTAR]({ commit }) {
    commit(GlobalMutationTypes.SET_LINK_PARA_VOLTAR, '');
  },
};

export default actions;
