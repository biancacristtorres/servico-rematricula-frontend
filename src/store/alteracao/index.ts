import { Module } from 'vuex';
import { AlteracaoState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';

const alteracao: Module<AlteracaoState, RootState> = {
  state,
  mutations,
  actions,
  getters,
  namespaced: true,
};

export default alteracao;
