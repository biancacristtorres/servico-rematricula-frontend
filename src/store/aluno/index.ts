import { Module } from 'vuex';
import { AlunoState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';


const Aluno: Module<AlunoState, RootState> = {
  actions,
  state,
  mutations,
  getters,
  namespaced: true,
};

export default Aluno;
