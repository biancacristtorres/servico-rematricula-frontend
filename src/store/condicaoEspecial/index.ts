import { Module } from 'vuex';
import { CondicaoEspecialState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';

const condicaoEspecial: Module<CondicaoEspecialState, RootState> = {
  state,
  mutations,
  actions,
  getters,
  namespaced: true,
};

export default condicaoEspecial;
