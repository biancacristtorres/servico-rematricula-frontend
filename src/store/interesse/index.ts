import { Module } from 'vuex';
import { InteresseState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';


const interesse: Module<InteresseState, RootState> = {
  actions,
  state,
  mutations,
  getters,
  namespaced: true,
};

export default interesse;
