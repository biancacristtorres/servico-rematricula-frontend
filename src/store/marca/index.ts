import { Module } from 'vuex';
import { state, MarcaState } from './state';
import { RootState } from '@/store';
import getters from './getters';

const marca: Module<MarcaState, RootState> = {
  state,
  getters,
  namespaced: true,
};

export default marca;
