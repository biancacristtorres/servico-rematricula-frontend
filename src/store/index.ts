import Vue from 'vue';
import Vuex from 'vuex';
import autorizacao from 'componente-frontend-core/store/autorizacao';
import interesse from './interesse';
import condicaoEspecial from './condicaoEspecial';
import featureToggle from 'componente-frontend-core/store/featureToggle';
import alteracao from './alteracao';
import resumo from './resumo';
import aluno from './aluno';
import actions from './actions';
import marca from './marca';
import miniGrade from './miniGrade';
import { state } from './state';
import mutations from './mutations';

export { StoreNamespaces } from '@/store/namespaces';

Vue.use(Vuex);

export class RootState {
}

const modules = {
  autorizacao,
  interesse,
  condicaoEspecial,
  featureToggle,
  alteracao,
  resumo,
  aluno,
  marca,
  miniGrade,
};

const store = new Vuex.Store({
  modules,
  actions,
  state,
  mutations,
});

Object.assign(window, { $store: store });

export default store;
