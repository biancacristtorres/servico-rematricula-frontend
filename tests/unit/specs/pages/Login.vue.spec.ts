import store from '@/store';
import login from '@/pages/login.vue';
import { shallowMount } from '@vue/test-utils';
import {
  LoginFactory,
} from '../../factory/pages/LoginFactory';

describe('Página de login', () => {
  const factory = new LoginFactory();

  it('Login invalido', async (done) => {
    factory.montarComponente();
    factory.dadoUmLoginInvalido();

    factory.vueInstance.entrar();

    expect(factory.componente).toMatchSnapshot();

    done();
  });

  it('Login válido', async (done) => {
    factory.componente = shallowMount(login, {
      localVue,
      store,
      sync: false,
      mocks: {
        $router: {
          push: (to: string) => {
            expect(to).toBe('/');
            done();
          },
        },
      },
    });

    factory.dadoUmLoginValido();
    factory.vueInstance.entrar();
    done();
  });
});
