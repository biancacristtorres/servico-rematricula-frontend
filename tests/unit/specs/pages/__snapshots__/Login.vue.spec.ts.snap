// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Página de login Login invalido 1`] = `
<vcontainer-stub tag="div" fluid="" fill-height="" grid-list-xl="">
  <vlayout-stub tag="div" justify-center="" align-center="">
    <vflex-stub tag="div" lg4="" md6="" sm8="" xs12="">
      <vcard-stub raised="true" tag="div" class="pa-5">
        <vform-stub>
          <valert-stub icon="warning" outline="true" transition="scale-transition" color="error" class="my-4">
            O R.A. ou senha preenchidos não foram encontrados no sistema. Favor corrigir e tentar novamente.
          </valert-stub>
          <vtextfield-stub data-vv-name="usuario" name="usuario" label="R.A." autocomplete="username" id="login-usuario"></vtextfield-stub>
          <vtextfield-stub data-vv-name="senha" name="senha" label="Senha" type="password" autocomplete="current-password" id="login-senha"></vtextfield-stub>
          <vcardactions-stub class="mt-4 justify-center pa-0">
            <vbtn-stub activeclass="v-btn--active" block="true" large="true" tag="button" type="button" id="login-botao" class="primary">
              Entrar
            </vbtn-stub>
          </vcardactions-stub>
        </vform-stub>
      </vcard-stub>
    </vflex-stub>
  </vlayout-stub>
</vcontainer-stub>
`;
